logstash/logstash.conf

```diff
@@ -63,11 +63,52 @@ filter{
   # }
   # mutate { add_field => { "[@metadata][target_index]" => "index-%{+YYYY.MM}" } }
 
+  grok {
+      match => { "message" => "%{GREEDYDATA:raw_message_content}" }
+  }
+  grok {
+      match => {
+        "message" => '<%{GREEDYDATA:skip} %{TIMESTAMP_ISO8601:skip} %{DATA:host} %{UUID:proc_id} \[%{DATA:instance_info}\] - \[tags@%{GREEDYDATA:skip} instance_id="%{INT:instance_id}" ip="%{IP:ip}" job="%{GREEDYDATA:skip} space_name="%{DATA:space_name}"\] %{GREEDYDATA:log_data}'
+        tag_on_failure => []
+        overwrite => ["message"]
+        break_on_match => false
+      }
+  }
+  # for message like: catalog-dev.data.gov - [2023-12-15T21:11:49.508628938Z] "GET /0000000 HTTP/1.1" 404 0 21445 "-" "Mozilla/5.0 ..." "127.0.x.x:xxxxx" "10.xx.2.10:xxxxx" x_forwarded_for:"108.xx.xxx.xxx, 64.252.66.xxx, 127.0.x.x" x_forwarded_proto:"https" ... ... b3:xxx
+  grok {
+      match => {
+        "log_data" => '%{HOSTNAME:hostname} - \[%{TIMESTAMP_ISO8601:timestamp}\] "%{WORD:http_method} %{GREEDYDATA:request} %{DATA:http_version}" %{NUMBER:status} %{NUMBER:bytes_sent} %{NUMBER:bytes_received} "%{DATA:http_referer}" "%{DATA:http_user_agent}" %{GREEDYDATA:skip} x_forwarded_for:"%{IP:real_ip}(?:, %{GREEDYDATA:forwarded_ips})*" x_forwarded_proto:%{GREEDYDATA:skip}'
+        tag_on_failure => []
+        overwrite => ["message"]
+        break_on_match => false
+      }
+  }
+  # for message like: 2023-12-14 20:31:53,839 INFO  [ckan.config.middleware.flask_app]  404 /dataset/xxxx render time 0.023 seconds
+  grok {
+      match => {
+        "log_data" => '%{TIMESTAMP_ISO8601:timestamp} %{LOGLEVEL:log_level}  \[%{DATA:logger}\]  %{NUMBER:status:int} %{URIPATH:request_path} render time %{NUMBER:render_time:float} seconds'
+        tag_on_failure => []
+        overwrite => ["message"]
+        break_on_match => false
+      }
+  }
+  if (
+      ([log_data] and [log_data] =~ /^NginxLog/)
```

Comment:
This will allow a hacker to hide his trace by adding a Nginxlog in the request.

Reply:
Never mind, it is safe, since hacker cant make the log message START with NginxLog.